clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -5438,8 +5438,20 @@ void Clang::ConstructJob(Compilation &C, const JobAction &JA,
   // Set the main file name, so that debug info works even with
   // -save-temps.
   CmdArgs.push_back("-main-file-name");
-  CmdArgs.push_back(getBaseInputName(Args, Input));
+  if (!IsSYCL || Args.hasArg(options::OPT_fno_sycl_use_footer)) {
+    CmdArgs.push_back(getBaseInputName(Args, Input));
+  } else {
+    SmallString<256> AbsPath = llvm::StringRef(Input.getBaseInput());
+    D.getVFS().makeAbsolute(AbsPath);
+    CmdArgs.push_back(
+        Args.MakeArgString(llvm::sys::path::filename(Input.getBaseInput())));
+    CmdArgs.push_back("-fsycl-use-main-file-name");
+  }
 
+  if (IsSYCL || Args.hasArg(options::OPT_fsycl_footer_path_EQ)) {
```

Comment:
Is the IsSYCL check needed here?

Reply:
Not sure. Does Args.hasArg(options::OPT_fsycl_footer_path_EQ) == true implies IsSYCL?

Reply:
I talked to Mike Rice and it looks like OpenMP doesn't use footers.  So I can presumably just check for the option and not test for IsSycl.